### ISSUE TITLE
NameError: name 'file' is not defined in Python3

### ISSUE THREAD

**Issue by pushlog** [telegraphic/hickle#41]
Used the below code to dump 100 x 100 numpy array to a hickle file

```
def save_hickle(file_name, data):
    hickle.dump(data,file_name,mode='w')


#save hickle
data = np.zeros([100, 100])
hickle_name ='data_100' + '.hkl'
save_hickle(hickle_name, data)
```

I get the following Traceback:

> NameError                                 Traceback (most recent call last)
> <ipython-input-39-5ea3b0d47216> in <module>()
>      27     #save hickle
>      28     hickle_name = filename + '.hkl'
> ---> 29     save_hickle(hickle_name, data)
>      30 
>      31     # save joblib
> 
> <ipython-input-38-6a9534d72751> in save_hickle(file_name, data)
>      18 
>      19 def save_hickle(file_name, data):
> ---> 20     hickle.dump(data,file_name,mode='w')
>      21 
>      22 def save_joblib(filename, data):
> 
> /Users/machinename/anaconda/lib/python3.5/site-packages/hickle.py in dump(py_obj, file_obj, mode, track_times, path, **kwargs)
>     306     try:
>     307         # Open the file
> --> 308         h5f = file_opener(file_obj, mode, track_times)
>     309         h5f.attrs["CLASS"] = 'hickle'
>     310         h5f.attrs["VERSION"] = 3
> 
> /Users/machinename/anaconda/lib/python3.5/site-packages/hickle.py in file_opener(f, mode, track_times)
>     146     """
>     147     # Were we handed a file object or just a file name string?
> --> 148     if isinstance(f, file):
>     149         filename, mode = f.name, f.mode
>     150         f.close()
> 
> NameError: name 'file' is not defined

**Comment by karandwivedi42** [comment]
I modified hickle so that it works in Python 3

https://github.com/karandwivedi42/hickle

**Comment by gidim** [comment]
Any chance @karandwivedi42 commit would be pulled to source? I'd love to have this on Python3

**Comment by scholer** [comment]
Also, if hickle does not support python 3, it would be prudent to add a list of supported versions to setup.py, to let pip/pypi know that hickle is python 2 only. Otherwise you are just wasting people's time and losing their respect and patience with this project.

**Comment by telegraphic** [comment]
Dear @gidim @scholer @karandwivedi42 and @pushlog -- I encourage you to try out the [dev branch](https://github.com/telegraphic/hickle/tree/dev) for preliminary Python 3 support. I took a different approach to @karandwivedi42 with a pretty major refactor that should make hickle more extensible in the long run. Remember that this branch is experimental!

@scholer as far as I can tell pip doesn't have any inbuilt Py2/Py3 checking mechanisms; in the master branch I have added in a manual check to exit the `setup.py` if you try and install on Python 3.

Closing this issue -- continued Py3 discussion may take place on https://github.com/telegraphic/hickle/issues/8
